src/tribler-core/tribler_core/tests/test_notifier.py

```diff
@@ -1,14 +1,20 @@
+from unittest.mock import Mock
+
+import pytest
+
 from tribler_common.simpledefs import NTFY
 
+from tribler_core.modules.tunnel.test_full_session.test_tunnel_community import deliver_messages
 from tribler_core.notifier import Notifier
 
 
-def test_notifier():
+@pytest.mark.asyncio
+async def test_notifier():
     notifier = Notifier()
 
-    def callback_func(*_):
-        callback_func.called = True
-
-    notifier.add_observer(NTFY.TORRENT_FINISHED, callback_func)
+    mock_foo = Mock()
+    notifier.add_observer(NTFY.TORRENT_FINISHED, mock_foo.bar)
     notifier.notify(NTFY.TORRENT_FINISHED)
-    assert callback_func.called
+    # Notifier uses asyncio loop internally, so we must wait at least a single loop cycle
+    await deliver_messages()
```

Comment:
I'm not sure if `deliver_messages` is the best way to skip a event loop cycle (it also looks that you are importing it from the tunnel tests?). Are there any alternatives provided by asyncio to achieve this?

Reply:
> Are there any alternatives provided by asyncio to achieve this?

If you only want to yield the current execution loop, you can use `asyncio.sleep(0)`.

Reply:
@qstokkink , thanks, I'll try that!